toolkit/tools/imagecustomizerapi/dirconfig.go

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//
+
+package imagecustomizerapi
+
+import (
+	"fmt"
+)
+
+// DirConfigList is a list of destination files where the source file will be copied to in the final image.
+// This type exists to allow a custom marshaller to be attached to it.
+type DirConfigList []DirConfig
+
+type DirConfig struct {
+	// The path in the target OS that the directory will be copied to.
+	SourcePath string `yaml:"SourcePath"`
+
+	// The path in the target OS that the directory will be copied to.
+	DestinationPath string `yaml:"DestinationPath"`
+
+	// The permissions to set on the directory.
+	Permissions *DirPermissions `yaml:"Permissions"`
+
+	// The permissions to set on the children of the directory.
+	ChildPermissions *DirPermissions `yaml:"ChildPermissions"`
```

Comment:
Looking at the implementation, `ChildPermissions` seems to the permissions applied to all the descendent files (that are copied) of the directory. So, perhaps rename it to `DescendentFilesPermissions`? Or `CopiedFilesPermissions`?

Reply:
I chose `childPermissions` since it is also being applied to the subdirectories being copied over, so specifically adding files to the name of this feild might be misleading

Reply:
Changed name